modules/Microsoft.Network/dnsZones/PTR/deploy.bicep

```diff
@@ -0,0 +1,68 @@
+@description('Conditional. The name of the parent DNS zone. Required if the template is used in a standalone deployment.')
+param dnsZoneName string
+
+@description('Required. The name of the PTR record.')
+param name string
+
+@description('Optional. The metadata attached to the record set.')
+param metadata object = {}
+
+@description('Optional. The list of PTR records in the record set.')
+param ptrRecords array = []
+
+@description('Optional. The TTL (time-to-live) of the records in the record set.')
+param ttl int = 3600
+
+@description('Optional. Array of role assignment objects that contain the \'roleDefinitionIdOrName\' and \'principalId\' to define RBAC role assignments on this resource. In the roleDefinitionIdOrName attribute, you can provide either the display name of the role definition, or its fully qualified ID in the following format: \'/providers/Microsoft.Authorization/roleDefinitions/c2f4ef07-c644-48eb-af81-4b1b4947fb11\'.')
+param roleAssignments array = []
+
+@description('Optional. Enable telemetry via a Globally Unique Identifier (GUID).')
+param enableDefaultTelemetry bool = true
+
+resource defaultTelemetry 'Microsoft.Resources/deployments@2021-04-01' = if (enableDefaultTelemetry) {
+  name: 'pid-47ed15a6-730a-4827-bcb4-0fd963ffbd82-${uniqueString(deployment().name)}'
+  properties: {
+    mode: 'Incremental'
+    template: {
+      '$schema': 'https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#'
+      contentVersion: '1.0.0.0'
+      resources: []
+    }
+  }
+}
+
+module PTR_roleAssignments '.bicep/nested_roleAssignments.bicep' = [for (roleAssignment, index) in roleAssignments: {
```

Comment:
Please move deployment below the main resource's deployment. Makes the file a bit more readable if it's ordered as per its dependencies

Reply:
Updated